python/pylibcudf/pylibcudf/libcudf/contiguous_split.pxd

```diff
@@ -1,19 +1,40 @@
-# Copyright (c) 2023-2024, NVIDIA CORPORATION.
+# Copyright (c) 2023-2025, NVIDIA CORPORATION.
 from libc.stdint cimport uint8_t
+from libc.stddef cimport size_t
+from libcpp cimport bool
 from libcpp.memory cimport unique_ptr
 from libcpp.vector cimport vector
 from pylibcudf.exception_handler cimport libcudf_exception_handler
 from pylibcudf.libcudf.table.table_view cimport table_view
 from pylibcudf.libcudf.types cimport size_type
-
+from pylibcudf.libcudf.utilities.span cimport device_span
 from rmm.librmm.device_buffer cimport device_buffer
+from rmm.librmm.cuda_stream_view cimport cuda_stream_view
+from rmm.librmm.memory_resource cimport device_memory_resource
 
 
 cdef extern from "cudf/contiguous_split.hpp" namespace "cudf" nogil:
     cdef cppclass packed_columns:
         unique_ptr[vector[uint8_t]] metadata
         unique_ptr[device_buffer] gpu_data
 
+    cdef cppclass chunked_pack:
+        bool has_next()
+        size_t get_total_contiguous_size()
+        size_t next(
+            const device_span[uint8_t] &
+        ) except +libcudf_exception_handler
+        unique_ptr[vector[uint8_t]] build_metadata(
+        ) except +libcudf_exception_handler
+
+        @staticmethod
+        unique_ptr[chunked_pack] create(
+            const table_view & input,
+            size_t user_buffer_size,
+            cuda_stream_view stream,
+            device_memory_resource *temp_mr,
```

Comment:
Is this how we're exposing  memory resources in general in pylibcudf? Ie. A pointer to a device memory resource.

Reply:
Ok gotcha. If you don't mind, can you add that as a comment?